dashboards-observability/public/components/visualizations/charts/lines/line.tsx

```diff
@@ -75,149 +86,189 @@ export const Line = ({ visualizations, layout, config }: any) => {
   if (!isEmpty(xaxis) && !isEmpty(yaxis)) {
     valueSeries = [...yaxis];
   } else {
-    valueSeries = (defaultAxes.yaxis || take(fields, lastIndex > 0 ? lastIndex : 1))
-      .map((item, i) => ({ ...item, side: i === 0 ? 'left' : 'right' }));
+    valueSeries = (
+      defaultAxes.yaxis || take(fields, lastIndex > 0 ? lastIndex : 1)
+    ).map((item, i) => ({ ...item, side: i === 0 ? 'left' : 'right' }));
   }
 
   const isDimensionTimestamp = isEmpty(xaxis)
     ? defaultAxes?.xaxis?.length && defaultAxes.xaxis[0].type === 'timestamp'
     : xaxis.length === 1 && xaxis[0].type === 'timestamp';
 
   let multiMetrics = {};
-  const [calculatedLayout, lineValues] = useMemo(() => {
-    const isBarMode = mode === 'bar';
-    let calculatedLineValues = valueSeries.map((field: any, index: number) => {
-      const selectedColor = getSelectedColorTheme(field, index);
-      const fillColor = hexToRgb(selectedColor, fillOpacity);
-      const barMarker = {
-        color: fillColor,
-        line: {
+  const isBarMode = mode === 'bar';
+  let calculatedLineValues = valueSeries.map((field: any, index: number) => {
```

Comment:
Any reason to move out of useMemo?

Reply:
We are facing issues to use useState() and as per discussion with @abasatwar useMemo() was not playing any significant role here, hence removed